app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -211,9 +249,155 @@ class HomeMenu(
+        }
+
+        val requestDesktopSiteItem = BrowserMenuImageSwitch(
+            imageResource = R.drawable.ic_desktop,
+            label = context.getString(R.string.browser_menu_desktop_site),
+            initialState = { false }
+        ) { checked ->
+            onItemTapped.invoke(Item.RequestDesktopSite(checked))
+        }
+
+        val whatsNewItem = BrowserMenuHighlightableItem(
+            context.getString(R.string.browser_menu_whats_new),
+            R.drawable.ic_whats_new,
+            iconTintColorResource = primaryTextColor,
+            highlight = BrowserMenuHighlight.LowPriority(
+                notificationTint = getColor(context, R.color.whats_new_notification_color)
+            ),
+            isHighlighted = { WhatsNew.shouldHighlightWhatsNew(context) }
+        ) {
+            onItemTapped.invoke(Item.WhatsNew)
+        }
+
+        val helpItem = BrowserMenuImageText(
+            context.getString(R.string.browser_menu_help),
+            R.drawable.ic_help,
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.Help)
+        }
+
+        val settingsItem = BrowserMenuImageText(
+            context.getString(R.string.browser_menu_settings),
+            R.drawable.ic_settings,
+            primaryTextColor
+        ) {
+            onItemTapped.invoke(Item.Settings)
+        }
+
+        // Only query account manager if it has been initialized.
+        // We don't want to cause its initialization just for this check.
+        val accountAuthItem =
+            if (context.components.backgroundServices.accountManagerAvailableQueue.isReady()
+                && context.components.backgroundServices.accountManager.accountNeedsReauth()) {
+                reconnectToSyncItem
+            } else {
+                null
+            }
+
+        val menuItems = listOfNotNull(
+            if (shouldUseBottomToolbar) null else menuToolbar,
+            if (settings.shouldDeleteBrowsingDataOnQuit) quitItem else null,
+            bookmarksItem,
+            historyItem,
+            downloadsItem,
+            extensionsItem,
+            syncSignInItem,
+            accountAuthItem,
+            BrowserMenuDivider(),
+            requestDesktopSiteItem,
+            BrowserMenuDivider(),
+            whatsNewItem,
+            helpItem,
+            settingsItem,
+            if (shouldUseBottomToolbar) BrowserMenuDivider() else null,
+            if (shouldUseBottomToolbar) menuToolbar else null
+        ).also { items ->
+            items.getHighlight()?.let { onHighlightPresent(it) }
+        }
+
+        return menuItems
+    }
+
     init {
+        val menuItems = if (FeatureFlags.toolbarMenuFeature) {
+            newCoreMenuItems()
+        } else {
+            oldCoreMenuItems
+        }
+
+        val menuItemsWithReconnectItem = if (FeatureFlags.toolbarMenuFeature) {
+            newCoreMenuItems() // re-evaluated with reconnect item viewable
```

Comment:
We need to re-evaluate the items here because we need to re-do the visibility of reconnect:

```kotlin
val accountAuthItem =
            if (context.components.backgroundServices.accountManagerAvailableQueue.isReady()
                && context.components.backgroundServices.accountManager.accountNeedsReauth()) {
                reconnectToSyncItem
            } else {
                null
            }
```

The new designs have this `accountAuthItem` in the middle of the list, so we can't just append it to the end like we used to. I'm taking this approach instead of trying to splice the list apart and add the auth item in the middle